llvm/lib/IR/DebugInfo.cpp

```diff
@@ -1686,7 +1686,8 @@ LLVMDbgRecordRef LLVMDIBuilderInsertDeclareRecordBefore(
   DbgInstPtr DbgInst = unwrap(Builder)->insertDeclare(
       unwrap(Storage), unwrap<DILocalVariable>(VarInfo),
       unwrap<DIExpression>(Expr), unwrap<DILocation>(DL),
-      unwrap<Instruction>(Instr));
+      Instr ? InsertPosition(unwrap<Instruction>(Instr)->getIterator())
+            : nullptr);
```

Comment:
Wouldn't this trip an assertion in the `nullptr` InsertPosition case (`assert(InsertPt.isValid());` in `insertDbgVariableRecord` called by `insertDeclare`). Or am I reading this wrong? (Same question applies to the other calls below)

Reply:
Thanks for unpicking that.

> This is a messy situation and in this PR I tried to just keep that exactly the same. What previously worked should still work. What previously didn't work should still not work.

SGTM. I feel as though these functions should probably have asserts in them saying as such but that could reasonably be argued to be out of scope of the patch.

Patch LGTM to me too.

Reply:
Thanks! I think for consistency, it would be nice to update these functions so that they all treat `nullptr` consistently, as a request to create whatever the function says to create but not insert it anywhere. When I have some extra time I'll see if that's easy to do.